src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -621,22 +631,24 @@ private[fastq] object FastqDemultiplexer {
     * @param numMismatches the # of mismatches if it matched a sample with a sample barcode.  This will be [[Int.MaxValue]]
     *                      for the unmatched sample.
     * @param records the records, one for each read that has template bases.
+    * @param passQc flag noting if the read passes QC. Default true to keep all reads unless otherwise specified.
+    * @param isControl flag noting if the read is an internal control. Default false unless filtering to remove internal control reads.
     */
-  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord], passQc: Boolean = true) {
+  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord], passQc: Boolean = true, isControl: Boolean = false) {
 
     /** Returns true if this [[DemuxResult]] should be kept for output, false otherwise.
      *
      * Returns true if `keepOnlyPassing` is false or if `passQc` is true
      *
      * @param keepOnlyPassing true to keep only passing reads, false to keep all reads
     */
-    def keep(keepOnlyPassing: Boolean = false): Boolean = {
-      if (!keepOnlyPassing) true else passQc
+    def keep(keepOnlyPassing: Boolean = false, keepControlReads: Boolean = false): Boolean = {
+      val keepByQc = if (!keepOnlyPassing) true else passQc
+      val keepByControl = if (keepControlReads) true else !isControl
+      keepByQc & keepByControl
```

Comment:
For booleans this isn't terrible, but `&` is bitwise and, and this should be using `&&` for logical and.